src/common/playlist/playlist.js

```diff
@@ -52,8 +52,9 @@ class Playlist {
     return {item: this._items[this._activeItemIndex] || null, index: this._activeItemIndex};
   }
 
-  get next(): {item: ?PlaylistItem, index: number} {
-    return {item: this._items[this._activeItemIndex + 1] || null, index: this._activeItemIndex + 1};
+  getNext(loop: boolean): {item: ?PlaylistItem, index: number} {
+    const nextIndex = loop ? (this._activeItemIndex + 1) % this._items.length : this._activeItemIndex + 1;
```

Comment:
doesn't playlist have a context of whether it's in loop or not? can we avoid passing this?

Reply:
@OrenMe  no. `loop` is `playlist-manager`s level option